File: tests/ext/things/TestTalker.js
TestTalker.doInit = function () {
  this.addTask("TestTask");
};

TestTalker.onTalk = function (talker) {
  this.giveTask(talker, "TestTask");
  talker.sendMessage("TestTalker heard you.\n");
};

TestTalker.onInspect = function (inspector) {
  return "A test NPC that offers TestTask.\n";
};

File: tests/ext/things/TestTalker.lua
function TestTalker:doInit()
    self:addTask("TestTask")
end

function TestTalker:onTalk(event)
    self:giveTask(event.actor, "TestTask")
    event.actor:sendMessage("TestTalker heard you.\n")
end

function TestTalker:onInspect(inspector)
    return "A test NPC that offers TestTask.\n"
end
